### ISSUE TITLE
I would like to import previous years' cost data into the dashboards

### ISSUE THREAD

**Issue by simonarbel** [microsoft/finops-toolkit#515]
Any idea how to do that?

**Comment by flanakin** [comment]
@simonarbel, is this about FinOps hubs? If so, we plan to add data retention and backfill support to get as much data as you want. For now, the simplest option is to update your export to a specific date range and then run it as a one-time export. FinOps toolkit 0.2 will include a PowerShell command that will help with this but you can do it manually in the portal today. We'll also add the auto-backfill support in either 0.3 or 0.4. I'll keep this around as a feature request for now.

**Comment by simonarbel** [comment]
Hey Michael

Yes this is regarding the Finops hub
AFAIK scheduled exporting cost data from the portal is only limited to 3 months back. Is there a way to go back earlier?

I tried to place usage data CSV  from previous year in the blob but that didn't work for me.



Simon



Get Outlook for Android<https://aka.ms/AAb9ysg>
________________________________

**Comment by flanakin** [comment]
The portal is limited to 3 months, but the API is not. We have an update coming to the `Start-FinOpsCostExport` PowerShell command that will do this in 0.2.1 and then a backfill feature for 0.3 that will solve it more explicitly.

```powershell
Start-FinOpsCostExport -Name {export-name} -Scope {scope-id}
```

**Comment by simonarbel** [comment]
Thanks so how this API can be used to populate the PBI dashboards?
Is there any explanation in the setup guide?

S

**Comment by flanakin** [comment]
No, we don't cover this today. I'll keep this issue open to add that to the docs. For now, you can use the Cost Management docs @ https://learn.microsoft.com/azure/cost-management-billing/automate/tutorial-seed-historical-cost-dataset-exports-api.

**Comment by simonarbel** [comment]
Thanks Michael
I will try to do a one time execution of this API to grab previous year's data in addition to the scheduled export ...but what should be the folder name in the storage container in that case?

Simon



Get Outlook for Android<https://aka.ms/AAb9ysg>
________________________________
